server/connectors/opensearch_connector/client.py

```
@@ -0,0 +1,169 @@
+        idx = index or self.index_pattern
+        return self._request("GET", f"/{idx}/_mapping")
+
+    @staticmethod
+    def normalize_endpoint(raw: str) -> str:
+        """Ensure endpoint has a scheme and no trailing slash."""
+        url = raw.strip().rstrip("/")
+        if not url.startswith(("http://", "https://")):
+            url = f"https://{url}"
+        parsed = urlparse(url)
+        if not parsed.netloc:
+            raise ValueError(f"Invalid OpenSearch endpoint: {raw!r}")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)